Restore with CSI volume failed with crash

**What steps did you take and what happened:**
<!--A clear and concise description of what the bug is, and what commands you ran.-->
The restore with CSI volume failed with crash, and the trace is:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x73b0ae]

goroutine 415 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x281c120?, 0x4727010?})
        /usr/local/go/src/runtime/panic.go:770 +0x132
k8s.io/apimachinery/pkg/api/resource.(*Quantity).ScaledValue(0xc0013de900?, 0x864640?)
        /go/pkg/mod/k8s.io/apimachinery@v0.29.0/pkg/api/resource/quantity.go:768 +0xe
k8s.io/apimachinery/pkg/api/resource.(*Quantity).Value(0xc0013de900?)
        /go/pkg/mod/k8s.io/apimachinery@v0.29.0/pkg/api/resource/quantity.go:754 +0x15
github.com/vmware-tanzu/velero/internal/volume.(*RestoreVolumeInfoTracker).Result(0xc001678690)
        /go/src/github.com/vmware-tanzu/velero/internal/volume/volumes_information.go:779 +0xb05
github.com/vmware-tanzu/velero/pkg/controller.(*restoreReconciler).runValidatedRestore(0xc000756180, 0xc0010ac2c8, {0xc000862388?, 0xc000bc8a80?}, 0x0)
        /go/src/github.com/vmware-tanzu/velero/pkg/controller/restore_controller.go:655 +0x2225
github.com/vmware-tanzu/velero/pkg/controller.(*restoreReconciler).Reconcile(0xc000756180, {0x3145040, 0xc0017ceb40}, {{{0xc0014ce696?, 0x0?}, {0xc001800390?, 0xc0017ceb40?}}})
        /go/src/github.com/vmware-tanzu/velero/pkg/controller/restore_controller.go:264 +0xb45
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x314d1f8?, {0x3145040?, 0xc0017ceb40?}, {{{0xc0014ce696?, 0xb?}, {0xc001800390?, 0x0?}}})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0005f0780, {0x3145078, 0xc00086e6e0}, {0x297d680, 0xc00032abc0})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:316 +0x3bc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0005f0780, {0x3145078, 0xc00086e6e0})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:266 +0x1be
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 259
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.2/pkg/internal/controller/controller.go:223 +0x50c
```

**What did you expect to happen:**
The restore should complete without error.

**The following information will help us better understand what's going on**:
The code here should check whether the size and status is nil before using.
https://github.com/vmware-tanzu/velero/blob/21366795d147348b66a31a9e905de202222c9492/internal/volume/volumes_information.go#L779

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`


**Anything else you would like to add:**
<!--Miscellaneous information that will assist in solving the issue.-->


**Environment:**

- Velero version (use `velero version`): v1.14.0-rc.2
- Velero features (use `velero client config get features`): EnableCSI
- Kubernetes version (use `kubectl version`): v1.29
- Kubernetes installer & version: EKS
- Cloud provider or hardware configuration: AWS
- OS (e.g. from `/etc/os-release`): Ubuntu


**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

--- a/internal/volume/volumes_information.go
+++ b/internal/volume/volumes_information.go
@@ -769,6 +769,14 @@ func (t *RestoreVolumeInfoTracker) Result() []*RestoreVolumeInfo {
 			continue
 		}
 		pvcNS, pvcName := n[0], n[1]
+		var restoreSize int64 = 0
+		if csiSnapshot.Status != nil && csiSnapshot.Status.RestoreSize != nil {
+			restoreSize = csiSnapshot.Status.RestoreSize.Value()
+		}
+		vscName := ""
+		if csiSnapshot.Spec.Source.VolumeSnapshotContentName != nil {
+			vscName = *csiSnapshot.Spec.Source.VolumeSnapshotContentName
+		}
 		volumeInfo := &RestoreVolumeInfo{
 			PVCNamespace:      pvcNS,
 			PVCName:           pvcName,
@@ -776,9 +784,9 @@ func (t *RestoreVolumeInfoTracker) Result() []*RestoreVolumeInfo {
 			RestoreMethod:     CSISnapshot,
 			CSISnapshotInfo: &CSISnapshotInfo{
 				SnapshotHandle: csiSnapshot.Annotations[VolumeSnapshotHandleAnnotation],
-				Size:           csiSnapshot.Status.RestoreSize.Value(),
+				Size:           restoreSize,
 				Driver:         csiSnapshot.Annotations[CSIDriverNameAnnotation],
-				VSCName:        *csiSnapshot.Spec.Source.VolumeSnapshotContentName,
+				VSCName:        vscName,
 			},
 		}
 		if pvcPVInfo := t.pvPvc.retrieve("", pvcName, pvcNS); pvcPVInfo != nil {

